Bottom: efe3c8ad24aac2c8efa16e23395752e21dfc1eac
Top:    a7d9767a8aa68774ee1aa112b3f63d61fea5dd2c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-20 23:47:21 +0100

Refresh of 3d-mouse-cursor-and-other-proj

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index fe2c11a..79ba618 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1503,13 +1503,15 @@ ghid_draw_everything (BoxTypePtr drawn_area)
 
   memset (do_group, 0, sizeof (do_group));
   for (ngroups = 0, i = 0; i < max_layer; i++) {
+    LayerType *l;
+    int group;
     int orderi;
 
     orderi = reverse_layers ? max_layer - i - 1 : i;
 
-    // Draw in numerical order
-    LayerType *l = LAYER_PTR (orderi); // LAYER_ON_STACK (i);
-    int group = GetLayerGroupNumberByNumber (orderi); // (LayerStack[i]);
+    // Draw in numerical order when in 3D view
+    l = global_view_2d ? LAYER_ON_STACK (i) : LAYER_PTR (orderi);
+    group = GetLayerGroupNumberByNumber (global_view_2d ? LayerStack[i] : orderi);
 
     if (/*l->On && */!do_group[group]) {
       do_group[group] = 1;
